### PR TITLE
Embeddings component

### PR DESCRIPTION
**Pull request by bhavani105**
Added an Embeddings component
Implementation of DPREmbeddings compoenent
Usage documentation in README

```Closes #(issue)```       

Notes:
- Replace `(issue)` **above** ↑↑↑ with the issue this PR closes to automatically link the two.
This must be done when the PR is created.
- Add multiple `Closes #(issue)` as needed.
- If this PR is work towards but does not close an issue, simply tag the issue without mentioning `Closes`.

## Description

**Describe** the changes proposed by this PR below to give the reviewer context below ↓↓↓

```(description)```     

### Request Review

Be sure to **request a review** from one or more reviewers (unless the PR is to an unprotected branch).

## Versioning

When opening a PR to make changes to PrimeQA (i.e. `primeqa/`) master, be sure to increment the version following
[semantic versioning](https://semver.org/).  The VERSION is stored [here](https://github.com/primeqa/primeqa/blob/main/VERSION) 
and is incremented using `bump2version {patch,minor,major}` as described in the (development guide documentation)[development.html].

 - [ ] Have you updated the VERSION?
 - [ ] Or does this PR not change the `primeqa` package or was not into master?

After pulling in changes from master to an existing PR, ensure the VERSION is updated appropriately.
This may require bumping the version again if it has been previously bumped.

If you're not quite ready yet to post a PR for review, feel free to open a [draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/).

### After Merging
If merging into master and VERSION was updated, after this PR is merged:
- [ ] [Create a release](https://docs.github.com/en/github/administering-a-repository/releasing-projects-on-github/managing-releases-in-a-repository#creating-a-release) from the master with version equal to [VERSION](https://github.com/primeqa/primeqa/blob/main/VERSION)
- [ ] Not merging to master or VERSION not updated

### Checklist

Review the following and mark as completed:
- [ ] [Tag an issue or issues](#what-does-this-pr-do) this PR addresses.
- [ ] Added [description](#description) of changes proposed.
- [ ] Review requested as [appropriate](#request-review).
- [ ] Version bumped as [appropriate](#Versioning).
- [ ] New classes, methods, and functions documented.
- [ ] Documentation for modified code is updated.
- [ ] Built documentation to confirm it renders as expected (see [here](development.md)).
- [ ] Code cleaned up and commented out code removed.
- [ ] Tests added to ensure all functionalities tested at >= 60% unit test coverage (see [here](development.md)).
- [ ] Code cleaned up and commented out code removed.
- [ ] Release created [as needed](#after-merging) after merging.

### CODE REVIEW

File: primeqa/components/README.md [hunk 1]
```diff
@@ -102,7 +102,7 @@ The reranker requires a ColBERT checkpoint. A checkpoint can be downloaded from
 
 Usage:
 ```
-    from primeqa.components.reranker.seq_classification_reranker import SeqClassificationReranker
+    from primeqa.components.reranker.colbert_reranker import ColBERTReranker
```
Comment (jdpsen) [inline primeqa/components/README.md:105]: Is this related to embedding PR or added by mistake ?

**Review by jdpsen — APPROVED** [review]
Left a few comments. Please check if appropriate .
Looks good to me otherwise.

File: primeqa/components/base.py [hunk 2]
```diff
@@ -57,6 +61,8 @@ def predict(
     ) -> Dict[str, List[Dict]]:
         pass
 
+    def get_embeddings(self, *args, **kwargs):
```
[the remark this answers is not in the document]
Comment (bhavani105) [reply primeqa/components/base.py]: Removed from base

File: primeqa/components/base.py [hunk 3]
```diff
@@ -156,6 +164,9 @@ def get_engine_type(self) -> str:
     def index(self, collection: Union[List[dict], str], *args, **kwargs):
         pass
     
+    def get_embeddings(self, *args, **kwargs):
```
[the remark this answers is not in the document]
Comment (bhavani105) [reply primeqa/components/base.py]: Removed

**Review by jsmcibm — APPROVED** [review]
Is it useful to have a mechanism for get_embeddings to directly return either Tensor and numpy array embeddings rather than json?  I guess this depends upon the use case, but I can imagine users converting the json beck to Tensor or numpy formats.
Otherwise LGTM

**Review by rongzhangibm — APPROVED** [review]
LGTM.

The only concern is that, the current script will generate the tokenization output as 
"title \<sep\> text" 
while <sep> is the separator used by tokenizer, e.g. </s> for Bart. 
In the case that title is none, this output may be worse than the tokenization of text only.
